libr/anal/p/anal_sh.c

```diff
@@ -269,87 +289,120 @@ static RAnalValue *anal_regrel_jump(RAnal* anal, RAnalOp* op, ut8 reg){
 		op->type = R_ANAL_OP_TYPE_LOAD;
 		op->src[0] = anal_fill_r0_reg_ref (anal, GET_SOURCE_REG(code), BYTE_SIZE);
 		op->dst = anal_fill_ai_rg (anal, GET_TARGET_REG(code));
+        r_strbuf_setf (&op->esil, "r0,r%d,+,[1],r%d,=,0x000000FF,r%d,&=,r%d,0x80,&,?{,0xFFFFFF00,r%d,|=,}",GET_SOURCE_REG(code),GET_TARGET_REG(code),GET_TARGET_REG(code),GET_TARGET_REG(code),GET_TARGET_REG(code));
 	} else if (IS_MOVW_R0REL_TO_REG(code)) {
 		op->type = R_ANAL_OP_TYPE_LOAD;
 		op->src[0] = anal_fill_r0_reg_ref (anal, GET_SOURCE_REG(code), WORD_SIZE);
 		op->dst = anal_fill_ai_rg (anal, GET_TARGET_REG(code));
+        r_strbuf_setf (&op->esil, "r0,r%d,+,[2],r%d,=,0x0000FFFF,r%d,&=,r%d,0x8000,&,?{,0xFFFF0000,r%d,|=,}",GET_SOURCE_REG(code),GET_TARGET_REG(code),GET_TARGET_REG(code),GET_TARGET_REG(code),GET_TARGET_REG(code));
 	} else if (IS_MOVL_R0REL_TO_REG(code)) {
 		op->type = R_ANAL_OP_TYPE_LOAD;
 		op->src[0] = anal_fill_r0_reg_ref (anal, GET_SOURCE_REG(code), LONG_SIZE);
 		op->dst = anal_fill_ai_rg (anal, GET_TARGET_REG(code));
+        r_strbuf_setf (&op->esil, "r0,r%d,+,[4],r%d,=",GET_SOURCE_REG(code),GET_TARGET_REG(code));
 	} else if (IS_NOP(code)) {
 		op->type = R_ANAL_OP_TYPE_NOP;
+        r_strbuf_setf (&op->esil, " ");
 	} else if (IS_CLRT(code)) {
-		op->type = R_ANAL_OP_TYPE_UNK;	//TODO : implement flag
+        op->type = R_ANAL_OP_TYPE_UNK;
+        r_strbuf_setf (&op->esil, "0xFFFFFFFE,sr,&=");
 	} else if (IS_SETT(code)) {
 		op->type = R_ANAL_OP_TYPE_UNK;
+        r_strbuf_setf (&op->esil, "0xFFFFFFFE,sr,|=");
 	} else if (IS_CLRMAC(code)) {
-		op->type = R_ANAL_OP_TYPE_UNK;	//TODO : type_mov ?
+        op->type = R_ANAL_OP_TYPE_UNK;
+        r_strbuf_setf (&op->esil, "0,mach,=,0,macl,=");
 	} else if (IS_DIV0U(code)) {
 		op->type = R_ANAL_OP_TYPE_DIV;
+        r_strbuf_setf (&op->esil, "0xFFFFFCFE,sr,|=");
 	} else if (IS_MOVT(code)) {
 		op->type = R_ANAL_OP_TYPE_MOV;
-		//op->src[0] = 		//TODO: figure out how to get T flag from sr reg
 		op->dst = anal_fill_ai_rg (anal, GET_TARGET_REG(code));
-	} else if (IS_MULL(code)) {
+        r_strbuf_setf (&op->esil, "0x1,sr,&,r%d,=",GET_TARGET_REG(code));
+    } else if (IS_MULL(code)) { //multiply long
 		op->type = R_ANAL_OP_TYPE_MUL;
 		op->src[0] = anal_fill_ai_rg (anal, GET_TARGET_REG(code));
 		op->src[1] = anal_fill_ai_rg (anal, GET_SOURCE_REG(code));
-		//op->dst = //TODO: figure out how to set MACL + MACH
+        r_strbuf_setf (&op->esil, "r%d,r%d,*,macl,=",GET_TARGET_REG(code),GET_SOURCE_REG(code));
 	} else if (IS_SLEEP(code)) {
 		op->type = R_ANAL_OP_TYPE_UNK;
-	} else if (IS_STSMAC(code)) {	//0000nnnn0000101_ sts MAC*,<REG_N>
+    } else if (IS_STSMACH(code)) {	//0000nnnn0000101_ sts MAC*,<REG_N>
 		op->type = R_ANAL_OP_TYPE_MOV;
 		op->dst = anal_fill_ai_rg (anal, GET_TARGET_REG(code));
-	} else if (IS_STCSR1(code)) {	//0000nnnn00010010 stc {sr,gbr,vbr,ssr},<REG_N>
+        r_strbuf_setf (&op->esil, "mach,r%d,=",GET_TARGET_REG(code));
+    } else if (IS_STSLMACL(code)){
```

Comment:
Space before (

Reply:
I don't see what you mean. Maybe space after comma? Before GET_TARGET_REG?

Reply:
Ah, alright. I'll change it in all macros. Will do like IS_STSLMACL (code)